### DISCUSSION TITLE
ValueError: [E090] Extension 'has_token' already exists on Doc. To overwrite the existing extension, set `force=True` on `Doc.set_extension`.

### DISCUSSION THREAD

**Discussion by anaconda121** [explosion/spaCy#8374] [Help: Coding & Implementations]
## How to reproduce the behaviour
I'm learning about custom attributes, and everytime I create my own custom attribute, I get a variant of the error: `"ValueError: [E090] Extension '{custom_attribute_title}' already exists on Doc. To overwrite the existing extension, set `force=True` on `{Doc,Span,Text.set_extension}`."`

Here is an example code:
```py

def has_token(doc, token_text):
    return token_text in[token.text for token in doc]

Doc.set_extension("has_token", method = has_token)

doc = nlp("The sky is blue")
print(doc._.has_token("blue"), " -> blue")
print(doc._.has_token("cloud"), " -> cloud")
```

When I add force = True to Doc.set_extension..., it works. I was wondering if I should just set force = True for all attributes I create  or is something wrong with the way I'm coding the attribute. Thanks!

## Info about spaCy

* **spaCy version:** 2.0.16
* **Platform:** Windows-10-10.0.19041-SP0
* **Python version:** 3.7.10
* **Models:** en_core_web_md, en_core_web_sm

**Comment by polm** [comment]
Modified slightly to include imports your code works fine for me.

```
import spacy
from spacy.tokens import Doc
nlp = spacy.blank("en")

def has_token(doc, token_text):
    return token_text in[token.text for token in doc]

Doc.set_extension("has_token", method = has_token)

doc = nlp("The sky is blue")
print(doc._.has_token("blue"), " -> blue")
print(doc._.has_token("cloud"), " -> cloud")
```

Can you give a complete code sample?

Do you have a loop somewhere? Usually that error means you are calling the `set_extension` method multiple times. You should only call it once per program.

**Reply by polm** [reply]
Also I mentioned this in your other thread but please upgrade your spaCy version.
